src/components/Accordion/Accordion.tsx

```diff
@@ -1,14 +1,54 @@
+import { AccordionVariantKey, AccordionVariants } from './types';
+
 import React from 'react';
 import classnames from 'classnames';
 import { getCommonProps } from '../../utils';
 import * as Accordion from '@radix-ui/react-accordion';
 
-// You'll need to change the HTMLDivElement to match the top-level element of your component
+interface AccordionVariantProps {
+  /**
+   * Determines whether multiple elements can be opened at the same time or not.
+   */
+  type: 'single' | 'multiple';
+  /**
+   * Determines if an open element can be closed by clicking on it.
+   * Only applicable to the `single` variants.
+   */
+  collapsible?: boolean;
+}
+
+/**
+ * Sets the type and collapsible props based on the variant prop
+ * The collapsible prop should only be passed when the variant is single
+ */
+const getAccordionVariantProps = (variant?: AccordionVariantKey): AccordionVariantProps => {
+  const variantProps: AccordionVariantProps = { type: 'multiple' };
+
+  if (variant === AccordionVariants.singleCollapsible) {
```

Comment:
I would be tempted to do this as a switch statement in a standalone function, not sure if you agree

Reply:
or at least extract it to a separately testable function and write a testcase specifically targeted at it.